### PR TITLE
Fix closing of datepicker on scroll

### PR DESCRIPTION
**Pull request by knight-bubble**
#2869

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://hirtie-maxim-fix-19612-date-picker-scroll-closing-2.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://hirtie-maxim-fix-19612-date-picker-scroll-closing-2.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by floriank** [comment]
Quick note for @hirtie-maxim:

I know it's daunting, but if you reopen a PR could you still add the link to the original ticket? WE had to search for it on Friday - and the current Github integration is not very powerful :sob: 

http://community.openproject.org/work_packages/19612

**Comment by floriank** [comment]
~~LGTM functionality wise.~~

Correction:
- I found an issue where the clear button is added twice:

screenshot from 2015-04-27 10 25 30
- The date parsing is still broken - I cannot use a date before Dec 2020 for some reason.
  
  @hirtie-maxim I guess 2. will be fixed in #2900 or #2904?

File: app/assets/stylesheets/content/_in_place_editing.sass [hunk 1]
```diff
@@ -47,14 +47,21 @@ $inplace-edit--color--very-dark: #cacaca
       cursor: wait!important
       inplace-edit--write
   .inplace-edit--write-value
-    .date-range-picker
+    .inplace-edit--date-range
```
Comment (myabc) [inline app/assets/stylesheets/content/_in_place_editing.sass]: @hirtie-maxim we generally avoid nesting selectors (unless you actually want the behaviour to differ in particular contexts)

File: frontend/app/ui_components/date/date-range-picker-directive.js [hunk 2]
```diff
@@ -195,31 +197,43 @@ module.exports = function(TimezoneService, ConfigurationService,
...
+          divEnd.show();
+          divStart.hide();
+        }
       });
 
-      angular.element(window).resize(function() {
-        inputStart.datepicker('hide');
-        inputEnd.datepicker('hide');
-        angular.element('#ui-datepicker-div').blur();
+      jQuery('.work-packages--details-content').on('click', function(e) {
```
Comment (myabc) [inline frontend/app/ui_components/date/date-range-picker-directive.js]: @hirtie-maxim let's standardise on `angular.element`? (or WDYT @0xF013?)

**Comment by myabc** [comment]
JS-code wide: looks good. CSS: see comments.

File: frontend/app/ui_components/date/date-range-picker-directive.js (hunk 2 above)
Comment (0xF013) [inline frontend/app/ui_components/date/date-range-picker-directive.js]: agree on angular.element

Comment (floriank) [inline frontend/app/ui_components/date/date-range-picker-directive.js]: jep, please use `angular.element`

File: app/assets/stylesheets/content/_in_place_editing.sass (hunk 1 above)
Comment (0xF013) [inline app/assets/stylesheets/content/_in_place_editing.sass]: > behaviour

spot the brit

**Comment by knight-bubble** [comment]
@myabc, can you take a look now at css, not sure what did you mean?

**Comment by floriank** [comment]
Maybe rebase this on the current `release/4.1` ?

**Comment by knight-bubble** [comment]
@floriank, Already did :smile:

**Comment by 0xF013** [comment]
ok, not hidden on scroll. the issue with two clear buttons is out of scope since another PR will only show one datepicker and that works fine so far

**Comment by 0xF013** [comment]
@myabc waiting for your word on CSS

**Comment by floriank** [comment]
LGTM - the datepicker stays open after a scroll.

**Pull request merged**